cmd/variable.go

```
@@ -54,45 +58,80 @@ var variableCreateCmd = &cobra.Command{
+					strct := s.MapIndex(key)
+					k := fmt.Sprintf("%s", key.Interface())
+					v := fmt.Sprintf("%s", strct.Interface())
+					varsFile = append(varsFile, fmt.Sprintf("%s=%s", k, v))
+				}
 			}
+
+			// Add it to the list of variables to create.
+			varOptions = append(varOptions, createVariableOptions(varsFile, tfe.CategoryTerraform, false, false)...)
+		}
+
+		// List existing variables.
+		variables, err := listVariables(client, workspace.ID)
+		if err != nil {
+			log.Fatalf("Cannot list the variables for  %q: %s.", organization, err)
+		}
+
+		// Index them by key.
+		indexedVars := map[string]*tfe.Variable{}
+		for _, v := range variables {
+			indexedVars[v.Key] = v
 		}
 
 		// Go through all the variables.
 		for _, opts := range varOptions {
-
-			// Update it if needed.
-			if force {
-				// Ensure the variable exists.
-				if v, ok := indexedVars[*(opts.Key)]; ok {
-					// If so, update it.
-					options := tfe.VariableUpdateOptions{
-						Key:       opts.Key,
-						Value:     opts.Value,
-						HCL:       opts.HCL,
-						Sensitive: opts.Sensitive,
-					}
-					if _, err := client.Variables.Update(context.Background(), workspace.ID, v.ID, options); err != nil {
-						log.Fatalf("Cannot update variable %q: %s.", v.Key, err)
-					}
-					break
+			//Check if the variable already exists.
+			v, exists := indexedVars[*(opts.Key)]
+
+			// Update an existing variable.
+			if exists && force {
+				options := tfe.VariableUpdateOptions{
+					Key:       opts.Key,
+					Value:     opts.Value,
+					HCL:       opts.HCL,
+					Sensitive: opts.Sensitive,
+				}
+				if _, err := client.Variables.Update(context.Background(), workspace.ID, v.ID, options); err != nil {
+					log.Fatalf("Cannot update variable %q: %s.", *(opts.Key), err)
 				}
+				log.Debugf("Variable %q updated successfully.", *(opts.Key))
+				continue
+			}
+
+			// Skip an existing variable.
+			if exists {
+				log.Warningf("Cannot create %q: variable already exists.", *(opts.Key))
```

I think this should return an error to the user and exit with an error code, or they may assume everything was updated when it wasn't.

---

That makes sense. I'll change it.